src/ansi-c/c_expr.h

```diff
@@ -200,4 +200,28 @@ inline side_effect_expr_overflowt &to_side_effect_expr_overflow(exprt &expr)
   return static_cast<side_effect_expr_overflowt &>(side_effect_expr);
 }
 
+/// \brief A base class for an expression that indicates the pre-function-call
+/// value of an expression passed as a parameter to a function
+class old_exprt : public unary_exprt
+{
+public:
+  explicit old_exprt(irep_idt id, exprt variable)
+    : unary_exprt(id, std::move(variable))
```

Comment:
I would fix the ID of `old_exprt` to `ID_old` -- I don't see an extension coming whereby you need several different kinds of "old expressions".

Reply:
Likewise, this isn't a "base class".